packages/services-soap/src/module.ts

```diff
@@ -0,0 +1,27 @@
+import { Mode } from '@seagull/mode'
+import { config } from '@seagull/seed'
+import { ContainerModule, interfaces } from 'inversify'
+import { SoapClientSupplier } from './mode/cloud'
+import { SoapClientSupplierPure } from './mode/pure'
+import { SoapClientSupplierSeed } from './mode/seed'
+
+export const soapDIModule = new ContainerModule(bind => {
+  bind<SoapClientSupplier>(SoapClientSupplier).toSelf()
+})
+
+export const module = new ContainerModule((bind: interfaces.Bind) => {
+  bind(SoapClientSupplier)
+    .toSelf()
+    .when(
+      () =>
```

Comment:
I would define this function outside of this, to make it more compact and easier to understand

Reply:
in this case i prefer the descriptability in place, as the read complexity is lower by
````Bind SoapClientSupplier to self when Mode is cloud, connected or edge.````
instead of
````SomethingToBeNamed is cloud, connected or edge. Bind SoapClientSupplier to self when Mode is somethingToBeNamed.````
Still you're not wrong and i'm basing this on the fact that it's very unlikely these lines will change anytime soon